src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -85,6 +93,44 @@ public FindCommand parse(String args) throws ParseException {
      */
     private Predicate<Person> combinePredicates(List<Predicate<Person>> predicateList) {
         return predicateList.stream()
-                            .reduce(x -> true, (predicate1, predicate2) -> predicate1.and(predicate2));
+                .reduce(x -> true, (predicate1, predicate2) -> predicate1.and(predicate2));
+    }
+
+    /**
+     * Used for parsing the input given by the user for finding with respect to salary
+     * @param input A string describing the condition for the salary of the person.
+     *              It should take the form of (comparator)(number), where the comparator is any of:
+     *              ">", "<", ">=", "<=", "="
+     *              Valid examples: >=5, <3.25, =6.00
+     *              Invalid examples: =>5, ==4.50, 6
+     *
+     * @return A Predicate which checks if the person passes the given condition as described in the input.
+     */
+    private Predicate<Person> parseSalaryComparison(String input) throws ParseException {
+        try {
+            String comparator = input.substring(0, 2);
+            if (comparator.equals(">=")) {
+                float value = Float.parseFloat(input.substring(2));
+                return new SalaryIsMoreThanEqualPredicate(value);
+            } else if (comparator.equals("<=")) {
+                float value = Float.parseFloat(input.substring(2));
+                return new SalaryIsLessThanEqualPredicate(value);
+            } else if (comparator.charAt(0) == '>') {
+                float value = Float.parseFloat(input.substring(1));
+                return new SalaryIsMoreThanPredicate(value);
+            } else if (comparator.charAt(0) == '<') {
+                float value = Float.parseFloat(input.substring(1));
+                return new SalaryIsLessThanPredicate(value);
+            } else if (comparator.charAt(0) == '=') {
+                float value = Float.parseFloat(input.substring(1));
+                return new SalaryIsEqualPredicate(value);
+            } else {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+            }
```

Comment:
Might want to consider turning this to switch-case, but not really that big of an issue.